signature/mechanism.go

```diff
@@ -2,120 +2,31 @@
 
 package signature
 
-import (
-	"bytes"
-	"fmt"
-
-	"github.com/mtrmac/gpgme"
-)
-
 // SigningMechanism abstracts a way to sign binary blobs and verify their signatures.
+// Each mechanism should eventually be closed by calling Close().
 // FIXME: Eventually expand on keyIdentity (namespace them between mechanisms to
 // eliminate ambiguities, support CA signatures and perhaps other key properties)
 type SigningMechanism interface {
-	// ImportKeysFromBytes imports public keys from the supplied blob and returns their identities.
-	// The blob is assumed to have an appropriate format (the caller is expected to know which one).
-	// NOTE: This may modify long-term state (e.g. key storage in a directory underlying the mechanism).
-	ImportKeysFromBytes(blob []byte) ([]string, error)
-	// Sign creates a (non-detached) signature of input using keyidentity
+	Close() error
+	// SupportsSigning returns nil if the mechanism supports signing, or an error message.
+	SupportsSigning() error
```

Comment:
Having an extra check which does not require you to get a key ID etc. is convenient for things like
```go

	if err := mech.SupportsSigning(); err != nil {
		t.Skipf("Signing not supported: %v", err)
	}
```
which this PR adds to tests, or perhaps to modify the UI of a tool (replace signing description in `--help` with a message saying that signing is not supported in this build).

We could, of course, _also_ define a `SigningNotSupported` error type, and return that from `Sign()`, for those who don’t care to check in advance but still want to detect the situation.